### PR TITLE
zenscroll Declaration File (@types/zenscroll)

### PR DESCRIPTION
**Pull request by HamedFathi**
I created `zenscroll` declaration file (@types/zenscroll)

You can find library in [zenscroll](https://zengabor.github.io/zenscroll/)

### CODE REVIEW

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
This needs tests

**Comment by HamedFathi** [comment]
@RyanCavanaugh  , I added `zenscroll-tests.ts` and fixed a problem in declaration file

File: types/zenscroll/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    intoView(elem: HTMLElement, duration?: number, onDone?: () => void): void;
+    center(elem: HTMLElement, duration?: number, offset?: number, onDone?: () => void): void;
+    stop(): void;
+    moving(): boolean;
+    getY(): number;
+    getTopOf(elem: HTMLElement): number;
+    createScroller(scrollContainer: HTMLElement, defaultDuration?: number, edgeOffset?: number): zenscroll;
+}
+
+declare var zenScroll: zenscroll;
```
Comment (RyanCavanaugh) [inline types/zenscroll/index.d.ts]: Spelling error

File: types/zenscroll/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+declare module "zenscroll" {
```
Comment (RyanCavanaugh) [inline types/zenscroll/index.d.ts]: This should follow the pattern as other UMD modules - make the containing file a module, and use `export as namespace zenscroll` to create the global

File: types/zenscroll/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// Type definitions for zenscroll 4.0.0
```
Comment (RyanCavanaugh) [inline types/zenscroll/index.d.ts]: Please add dtslint.json to this PR as requested in the template

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/zenscroll/index.d.ts (hunk 3 above)
Comment (andy-ms) [reply types/zenscroll/index.d.ts]: Should have a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. (not `dtslint.json`)

**Comment by RyanCavanaugh** [comment]
@HamedFathi Please address comments from the code reviewers.

**Comment by HamedFathi** [comment]
@RyanCavanaugh , Unfortunately, I did not find a solution to the problems. So, I will close this request soon if nobody help me. I created the new one but it fails. I can not do more.

```
// Type definitions for zenscroll 4.0.0
// Project: https://zengabor.github.io/zenscroll/
// Definitions by: Hamed Fathi <https://github.com/HamedFathi/>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

declare namespace ZenScroll {
    interface setupOption {
        defaultDuration: number;
        edgeOffset: number;
    }

    interface zenscroll {
        setup(defaultDuration?: number, edgeOffset?: number): setupOption;
        to(elem: HTMLElement, duration?: number, onDone?: () => void): void;
        toY(targetY: number, duration?: number, onDone?: () => void): void;
        intoView(elem: HTMLElement, duration?: number, onDone?: () => void): void;
        center(elem: HTMLElement, duration?: number, offset?: number, onDone?: () => void): void;
        stop(): void;
        moving(): boolean;
        getY(): number;
        getTopOf(elem: HTMLElement): number;
        createScroller(scrollContainer: HTMLElement, defaultDuration?: number, edgeOffset?: number): zenscroll;
    }
}

declare var zenscroll: ZenScroll.zenscroll;
declare var zenscrollOptions: ZenScroll.setupOption;

export { zenscroll, zenscrollOptions };
export as namespace ZenScroll;
```

but I used it and failed

```
import * as zen from 'zenscroll';
    scrollUp() {
        var ctrl = document.getElementById("scroll");
        if (ctrl)
            zen.zenscroll.to(ctrl); // >>> undefined
    }



```

**Comment by RyanCavanaugh** [comment]
@RyanCavanaugh - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

File: types/zenscroll/index.d.ts (hunk 3 above)
Comment (HamedFathi) [reply types/zenscroll/index.d.ts]: I added it no problem now

File: types/zenscroll/index.d.ts (hunk 1 above)
Comment (HamedFathi) [reply types/zenscroll/index.d.ts]: Everything works no spelling error I checked it.

File: types/zenscroll/index.d.ts (hunk 2 above)
Comment (HamedFathi) [reply types/zenscroll/index.d.ts]: I Added 👍

**Comment by HamedFathi** [comment]
@RyanCavanaugh , If I can help, it will be proud of me. thanks for your efforts.
I did not find any problem, If you see something let me know. As I mentioned I checked all samples by this `.d.ts` based on [zenscroll](https://zengabor.github.io/zenscroll/) tutorial.

I think you can accept this pull request now! :)

**Pull request merged**